### PR TITLE
Remove Tooltip mouse tracker listener & update hovering/MouseRegion logic & animation

### LINKED ISSUE #117627: [BUG] Tooltip calls build when mouse enters/leaves Flutter window
## Bug Summary

When mouse enters/leaves the flutter app window,  `Tooltip` widget rebuilds its child unnecessarily. The mouse pointer doesn't even need to enter the `Tooltip` area, just by entering or leaving the app window, all `Tooltip` widgets rebuild their children.

## Demonstration

Sample code is provided below. There's  a `print()` in the `builder` which is the child of the `Tooltip`. Each time mouse enters or leaves the app, the `print()` is getting called and logs "rebuilding!". (My original code had `TweenAnimationBuilder` and it was repeating the animation over and over but I've stripped it down to a simple builder to demonstrate.)

https://user-images.githubusercontent.com/17809451/209461262-f571fc35-ef58-4177-8100-855f4da72f0f.mp4

## Sample code
```
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Tooltip bug'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Container(
          decoration:
              BoxDecoration(border: Border.all(width: 1, color: Colors.blue)),
          child: SizedBox(
            height: 100,
            width: 100,
            child: Tooltip(
              message: 'Some label',
              child: Builder(
                builder: (BuildContext context) {
                  // ignore: avoid_print
                  print('rebuilding!');
                  return const Icon(
                    Icons.abc_outlined,
                    size: 100,
                    color: Colors.red,
                  );
                },
              ),
            ),
          ),
        ),
      ),
    );
  }
}

```

## Flutter Doctor
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.3.10, on Microsoft Windows [Version 10.0.22621.963], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.3.6)
[√] Android Studio (version 2021.2)
[√] VS Code, 64-bit edition (version 1.74.1)
[√] Connected device (4 available)
[√] HTTP Host Availability

• No issues found!

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
Fixes https://github.com/flutter/flutter/issues/117627

### Behavior changes:
1. If fade in/fade out animation is already in progress, hiding/showing the tooltip will immediately take effect without waiting for `waitDuration`.
2. A PointerDownEvent that doesn't become a part of a "trigger" gesture dismisses the tooltip, even for hovered ones.
3. The OverlayEntry is now updated only when the previous tooltip was completely dismissed. This can be fixed by OverlayPortal but I'm not sure what the correct behavior is.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/material/checkbox_test.dart [hunk 1]
```diff
@@ -1577,9 +1577,12 @@ void main() {
       )
     );
 
-    final Finder tooltip1 = find.byType(Tooltip);
+    await tester.pump(const Duration(days: 1));
```
Comment (LongCatIsLooong) [inline packages/flutter/test/material/checkbox_test.dart:1580]: These tests were assuming the OverlayEntry tooltip updates when the `Tooltip` widget updates so I had to add code to wait for the previous tooltip to dismiss.

**Comment by gspencergoog** [comment]
The _revealLastTooltip was to deal with nested widgets which each had a tooltip on them so that only the deepest one would be shown.  See the before/after videos in this PR:
https://github.com/flutter/flutter/pull/90457

**Comment by LongCatIsLooong** [comment]
Ah because MouseRegion doesn't notify you when you move from the inner MouseRegion to the outer one?

**Comment by gspencergoog** [comment]
> Ah because MouseRegion doesn't notify you when you move from the inner MouseRegion to the outer one?

Yes.

**Comment by LongCatIsLooong** [comment]
Hi @gspencergoog would you like to take a look at this one when you get a chance?

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

This looks a lot better.

File: packages/flutter/lib/src/material/tooltip.dart [hunk 2]
```diff
@@ -19,6 +19,70 @@ import 'tooltip_visibility.dart';
 /// Signature for when a tooltip is triggered.
 typedef TooltipTriggeredCallback = void Function();
 
+// A special [MouseRegion] that when nested, only the first [_ExclusiveMouseRegion]
```
[the remark this answers is not in the document]
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/material/tooltip.dart:22]: Do you mean using `///` instead of `//`?

[the remark this answers is not in the document]
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/material/tooltip.dart]: Changed to `///`.

**Comment by LongCatIsLooong** [comment]
Setting Google Testing to passing: the failures were expected, LGTM'ed by internal customer.

**Pull request merged**